NStack/strings/ustring.cs

```diff
@@ -1187,17 +1187,14 @@ public int LastIndexOfAny (ustring chars)
 		}
 
 		/// <summary>
-		/// Implements the IComparable.CompareTo method
+		/// Implements the IComparable<typeparamref name="T"/>.CompareTo method
 		/// </summary>
 		/// <returns>Less than zero if this instance is less than value, zero if they are the same, and higher than zero if the instance is greater.</returns>
-		/// <param name="value">Value.</param>
-		public int CompareTo (object value)
+		/// <param name="other">Value.</param>
+		public int CompareTo (ustring other)
 		{
-			if (value == null)
-				return 1;
-			var other = value as ustring;
 			if (other == null)
-				throw new ArgumentException ("Argument must be a ustring");
```

Comment:
This check needs to be ((object) other == null) otherwise this triggers a call to operator ==.